provider/aws/aws.go

```diff
@@ -653,9 +711,65 @@ func (p *AWSProvider) newChange(action string, ep *endpoint.Endpoint, recordsCac
 		change.ResourceRecordSet.HealthCheckId = aws.String(prop.Value)
 	}
 
+	setOwnedRecord(change, ep)
+
 	return change, dualstack
 }
 
+// HACK: we have no easy way of passing the "ownedRecord" label of endpoints to `submitChanges` for grouping
+// transactions by owner relationship; we cheat here by passing the label as the `AliasTarget` of the record;
+// the `AliasTarget` setting is recognized and removed in `submitChanges` before submission of the change to Route53;
+// the "ownedRecord" label should only ever appear on TXT endpoints created by the TXT registry, so it should not
+// interfere with any other record that could possibly have an actual `AliasTarget` setting
+func setOwnedRecord(change *route53.Change, ep *endpoint.Endpoint) {
+	if ownedRecord, ok := ep.Labels[endpoint.OwnedRecordLabelKey]; ok && change.ResourceRecordSet.AliasTarget == nil {
+		change.ResourceRecordSet.AliasTarget = &route53.AliasTarget{
+			HostedZoneId: aws.String(endpoint.OwnedRecordLabelKey),
+			DNSName:      aws.String(ownedRecord),
+		}
+	}
+}
```

Comment:
What alternatives do we have to avoid this hack?

Reply:
The only alternative I was seeing was to wrap the `route53.Change` type in order to add a new field, but this would require pulling the change through _all_ methods and tests that handle the changes. Also, the `ownedRecord` field would semantically belong more to the record struct than to the change struct, but we can't wrap the record struct because it's nested in the change struct, so this would probably cause even more confusion.

Reply:
Please let me know what you prefer.

Reply:
I am not sure either, would it be too much effort to see how the `route53.Change` wrapping looks like? I don't think we should merge this hack as it is and that we must find a better way.

Reply:
I'll try to prepare something for this.

Reply:
@devkid any update on this one?

Reply:
@Raffo rebased and did the changes to wrap `route53.Change`. Let me know what you think of this.

Sorry for the _very_ late response here, I was lacking the time and motivation to continue this, both should be fixed now :)